### ISSUE TITLE
Wrong gradient calculated for API tf.keras.layers.Add

### ISSUE THREAD

**Issue by cheyennee** [tensorflow/tensorflow#62420]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

tf 2.14.0

### Custom code

Yes

### OS platform and distribution

windows colab

### Python version

3.10

### Current behavior?

Forward mode differentiation for the case is inconsistent with the gradient calculated in reverse mode. They shoule be the same.

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
inputs_0_tensor = tf.random.uniform([], minval=0.3, maxval=1, dtype=tf.float32)
inputs_0 = tf.identity(inputs_0_tensor)
inputs_1_tensor = tf.random.uniform([], minval=1.1, maxval=10,dtype=tf.float32)
inputs_1 = tf.identity(inputs_1_tensor)
inputs = [inputs_0, inputs_1]
layer = tf.keras.layers.Add()

with tf.GradientTape(persistent=True, ) as g:
    g.watch(inputs)
    res_backward = layer(inputs)
grad_backward = g.jacobian(res_backward, res_backward)
print("res_backward:", res_backward)
print("grad_backward:", grad_backward)
tangent_1 = tf.constant(1,dtype=tf.float32)
tangent_2 = tf.constant(1,dtype=tf.float32)
tangents = [tangent_1, tangent_2]

with tf.autodiff.ForwardAccumulator(inputs, tangents) as acc:
    res_forward = layer(inputs)
    grad_jvp = acc.jvp(res_forward)
    print("res_forward:", res_forward)
    print("grad_forward", grad_jvp)
```

### Relevant log output

```shell
res_backward: tf.Tensor(9.794308, shape=(), dtype=float32)
grad_backward: tf.Tensor(1.0, shape=(), dtype=float32)
res_forward: tf.Tensor(9.794308, shape=(), dtype=float32)
grad_forward tf.Tensor(2.0, shape=(), dtype=float32)
```

**Comment by Venkat6871** [comment]
Hi **@cheyennee** ,

I was able to reproduce the issue on colab using TF v2.14, 2.15, and TF-nightly. Please find the [gist](https://colab.research.google.com/gist/Venkat6871/9def4c1e50ace380e2e05d32980b89ab/62420_2-14_2-15-nightly_.ipynb) here for reference.

Thank you!

**Comment by LakshmiKalaKadali** [comment]
Hi @cheyennee,

Please specify the arguments for Jacobian function as res_backward and inputs like `grad_backward = g.jacobian(res_backward, inputs)` instead of passing same argument( res_backward) to the Jacobian function. Please refer to the [gist](https://colab.sandbox.google.com/gist/LakshmiKalaKadali/5b87f2dfa18345cc8de7e8d4de7f0916/tf_62420.ipynb#scrollTo=ceLeUb4Pg7HK). 

The issue here is that in the `reverse-mode`,  the gradients  are computed for each input independently, which results in the gradients [1.0, 1.0]. The same thing you can observe here in the `grad_backward` output:`grad_backward: [<tf.Tensor: shape=(), dtype=float32, numpy=1.0>, <tf.Tensor: shape=(), dtype=float32, numpy=1.0>]` which is expected behavior.
while forward-mode computes the total of derivatives, which is the sum of the gradients weighted by the tangents. Since your tangents are [1.0, 1.0], the result is the sum of the individual gradients: 2.0.

Thank You
